compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/BallerinaSemanticModel.java

```diff
@@ -183,40 +183,84 @@ public BallerinaSemanticModel(BLangPackage bLangPackage, CompilerContext context
      */
     @Override
     public List<Location> references(Document sourceDocument, LinePosition position) {
-        return references(sourceDocument, position, true);
+        BSymbol symbolAtCursor = findSymbolAtCursorPosition(sourceDocument, position);
+        if (symbolAtCursor == null) {
+            return Collections.emptyList();
+        }
+        Location symbolLocation = symbolAtCursor.getPosition();
+        BLangNode node = findEnclosingContainerNode(symbolLocation.lineRange(), null);
+        return getReferences(symbolAtCursor, node, true);
     }
 
     @Override
     public List<Location> references(Symbol symbol, boolean withDefinition) {
+        BSymbol symbolAtCursor = getInternalSymbol(symbol);
         Optional<Location> symbolLocation = symbol.getLocation();
+        if (symbolAtCursor == null || symbolLocation.isEmpty()) {
+            return Collections.emptyList();
+        }
```

Comment:
A couple of points. 
- Can check whether there's a location as the first check since there's no point in getting the internal symbol if there isn't a location.
- I think there's no need for a null check for `symbolAtCursor`. Every `Symbol` instance should have an associated internal symbol.